google/cloud/internal/oauth2_external_account_credentials.cc

```diff
@@ -149,12 +181,34 @@ StatusOr<internal::AccessToken> ExternalAccountCredentials::GetToken(
             .WithMetadata("token_type", *token_type)
             .WithMetadata("issued_token_type", *issued_token_type));
   }
+  if (info_.impersonation_config.has_value()) {
+    return GetTokenImpersonation(*token, ec);
+  }
+
   auto expires_in =
       ValidateIntField(access, "expires_in", "token-exchange-response", ec);
   if (!expires_in) return std::move(expires_in).status();
   return internal::AccessToken{*token, tp + std::chrono::seconds(*expires_in)};
 }
 
+StatusOr<internal::AccessToken>
+ExternalAccountCredentials::GetTokenImpersonation(
+    std::string const& access_token, internal::ErrorContext const& ec) {
+  auto request = rest_internal::RestRequest(info_.impersonation_config->url);
+  request.AddHeader("Authorization", "Bearer " + access_token);
```

Comment:
nit: do we care to `absl::StrCat()` ? (we did in `AuthorizationHeader(...)`)

Reply:
I cannot say that I mind a single `+` that much, but done.